flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestBase.java

```diff
@@ -71,6 +74,8 @@
  * <i>bc6b2b2d5f6424d5f377aa6c0871e82a956462ef</i>
  */
 @SuppressWarnings("serial")
+// Tests inheriting from this class are known to be unstable due to the test setup.
+@RetryOnFailure(times = 3)
```

Comment:
Thanks for your contribution! It looks overall great!

It looks like there are no @Test methods in the Base class. May I know the purpose of having the retry rule and logic here if every subclass has to define their own?

Reply:
Very good catch, I forgot that the annotations do not automatically apply to subclasses. I added 3cc1e3b so that this annotation is also discovered on super classes.

Reply:
I also updated the javadoc. PTAL